addon/model.js

```diff
@@ -1,39 +1,52 @@
 import DS from 'ember-data';
 import Ember from 'ember';
 
-const { attr, belongsTo, hasMany } = DS;
 const { camelize, pluralize } = Ember.String;
 
 function cleanKey(key) {
   return camelize(key.replace(/_id$|_ids$/, ''));
 }
 
+function buildProperty(key, name, type) {
+  if (type.match(/string|number|boolean|date/i)) {
+    if (key.match(/_id$/)) {
+      return [name, DS.belongsTo({ async: true })];
+    } else {
+      return [name, DS.attr(type)];
+    }
+  } else if (type === 'array') {
+    if (key.match(/_ids$/)) {
+      return [pluralize(name), DS.hasMany({ async: true })];
+    } else {
+      return [name, DS.attr({ defaultValue: [] })];
+    }
+  } else if (type === 'object') {
+    return [name, DS.attr({ defaultValue: {} })];
+  }
+}
+
 export default class Model {
   static generate(schema) {
-    const attrs = {};
+    const attributes = {};
 
     Object.keys(schema.properties).forEach(key => {
       const name = cleanKey(key);
       const property = schema.properties[key];
       const type = property.type;
 
-      if (type.match(/string|number|boolean|date/i)) {
-        if (key.match(/_id$/)) {
-          attrs[name] = belongsTo({ async: true });
-        } else {
-          attrs[name] = attr(type);
-        }
-      } else if (type === 'array') {
-        if (key.match(/_ids$/)) {
-          attrs[pluralize(name)] = hasMany({ async: true });
-        } else {
-          attrs[name] = attr({ defaultValue: [] });
-        }
-      } else if (type === 'object') {
-        attrs[name] = attr({ defaultValue: {} });
+      if (Ember.isArray(type)) {
+        const [firstType] = type;
+
+        const [attr, value] = buildProperty(key, name, firstType);
+
+        attributes[attr] = value;
+      } else {
+        const [attr, value] = buildProperty(key, name, type);
```

Comment:
const 'attr' has already been declared.<br>'attr' is already defined.<br>const 'value' has already been declared.<br>'value' is already defined.